HIP/hip-1056.md

```diff
@@ -0,0 +1,1952 @@
+repeated here. The full and detailed specification is present in the public
+[hedera-protobufs github](https://github.com/hashgraph/hedera-protobufs/tree/main/block) repository.
+
+#### Block
+
+A collection of block items that represent a single block of transactions on the network.
+
+```protobuf
+/**
+ * Block is used to serialize all the data for a block into record stream of
+ * block files. This structure represents a block on Hedera.
+ */
+message Block {
+  /**
+  * The items contained in the block that make up the running hash.
+  */
+  repeated BlockItem items = 1;
+}
+```
+
+The order of block items in a block stream is important and in some cases it designates useful positions in the stream that
+a client may make inferences on. Particularly the start of a `Block`, `Event` and `Transaction` are useful
+considerations, in addition to the end of a `Block` or `Transaction`.
+
+#### BlockItem
+
+A block item is a logical separation of components that together make up a block. A Block item may be one of the 10 types
+noted below.
+
+```protobuf
+message BlockItem {
+  oneof item {
+    /**
+    * An header for the block, marking the start of a new block.
+    */
+    com.hedera.hapi.block.stream.output.BlockHeader block_header = 1;
+
+    /**
+    * An header emitted at the start of a new network "event".
+    */
+    com.hedera.hapi.block.stream.input.EventHeader event_header = 2;
+
+    /**
+    * An header emitted at the start of a new consensus "round".
+    * <p>
+    * This item SHALL contain the properties relevant to a single
+    * consensus round.
+    */
+    com.hedera.hapi.block.stream.input.RoundHeader round_header = 3;
+
+    /**
+    * A single transaction.
+    */
+    com.hedera.hapi.platform.event.EventTransaction event_transaction = 4;
+
+    /**
+    * The result of running a transaction.
+    */
+    com.hedera.hapi.block.stream.output.TransactionResult transaction_result = 5;
+
+    /**
+    * A transaction output.
+    */
+    com.hedera.hapi.block.stream.output.TransactionOutput transaction_output = 6;
+
+    /**
+      * A set of state changes.
+      */
+    com.hedera.hapi.block.stream.output.StateChanges state_changes = 7;
+
+    /**
+    * Verification data for an item filtered from the stream.<br/>
+    * This is a hash for a merkle tree node where the contents of that
+    * part of the merkle tree have been removed from this stream.
+    */
+    FilteredItemHash filtered_item_hash = 8;
+
+    /**
+    * A signed block proof.<br/>
+    * The signed merkle proof for this block. This will validate
+    * a "virtual" merkle tree containing the previous block "virtual"
+    * root, an "input" subtree, an "output" subtree, and
+    * a "state changes" subtree.
+    */
+    BlockProof block_proof = 9;
+
+    /**
+    * A record file and associated data.
+    */
+    RecordFileItem record_file = 10;
+  }
+}
```

Comment:
The "fixed" indent is using the wrong settings:
1. indentation should be 4 spaces, not 2
1. comments should line up on `*`, so every line after the first should be indented one extra space.

### Example
```protobuf
        /**
         * A signed block proof.<br/>
         * The signed merkle proof for this block. This will validate
         * a "virtual" merkle tree containing the previous block "virtual"
         * root, an "input" subtree, an "output" subtree, and
         * a "state changes" subtree.
         */
        BlockProof block_proof = 9;

        /**
         * A record file and associated data.
         */
        RecordFileItem record_file = 10;
    }
}
```